WordPress/Classes/ViewRelated/Blog/Site Settings/HomepageSettingsViewController.swift

```diff
@@ -0,0 +1,175 @@
+
+    fileprivate var postService: PostService!
+
+    fileprivate lazy var handler: ImmuTableViewHandler = {
+        return ImmuTableViewHandler(takeOver: self)
+    }()
+
+    /// Designated Initializer
+    ///
+    /// - Parameter blog: The blog for which we want to configure Homepage settings
+    ///
+    @objc public convenience init(blog: Blog) {
+        self.init(style: .grouped)
+
+        self.blog = blog
+
+        let context = blog.managedObjectContext ?? ContextManager.shared.mainContext
+        postService = PostService(managedObjectContext: context)
+    }
+
+    open override func viewDidLoad() {
+        super.viewDidLoad()
+
+        title = Strings.title
+        clearsSelectionOnViewWillAppear = false
+
+        // Setup tableView
+        WPStyleGuide.configureColors(view: view, tableView: tableView)
+        WPStyleGuide.configureAutomaticHeightRows(for: tableView)
+
+        ImmuTable.registerRows([CheckmarkRow.self, NavigationItemRow.self, ActivityIndicatorRow.self], tableView: tableView)
+        reloadViewModel()
+
+        fetchAllPages()
+    }
+
+    private func fetchAllPages() {
+        let options = PostServiceSyncOptions()
+        options.number = 20
+
+        postService.syncPosts(ofType: .page, with: options, for: blog, success: { posts in
+            self.reloadViewModel()
+        }, failure: { _ in
+
+        })
+    }
+
+    // MARK: - Model
+
+    fileprivate func reloadViewModel() {
+        handler.viewModel = tableViewModel
+    }
+
+    fileprivate var tableViewModel: ImmuTable {
+        guard let homepageType = blog.homepageType else {
+            return ImmuTable(sections: [])
+        }
+
+        let changeTypeSection = ImmuTableSection(headerText: nil,
+                                                 rows: updating ? updatingHomepageTypeRows : homepageTypeRows,
+                                                 footerText: Strings.footerText)
+
+        let choosePagesSection = ImmuTableSection(headerText: Strings.choosePagesHeaderText, rows: selectedPagesRows)
+
+        var sections = [changeTypeSection]
+        if homepageType == .page {
+            sections.append(choosePagesSection)
+        }
+
+        return ImmuTable(sections: sections)
+    }
+
+    var updatingHomepageTypeRows: [ImmuTableRow] {
+        guard let homepageType = blog.homepageType else {
+            return []
+        }
+
+        return [
+            ActivityIndicatorRow(title: HomepageType.posts.title, animating: homepageType == .posts, action: nil),
+            ActivityIndicatorRow(title: HomepageType.page.title, animating: homepageType == .page, action: nil)
+        ]
+    }
+
+    var homepageTypeRows: [ImmuTableRow] {
+        guard let homepageType = blog.homepageType else {
+            return []
+        }
+
+        return [
+            CheckmarkRow(title: HomepageType.posts.title, checked: homepageType == .posts, action: { _ in
+                self.setHomepageType(.posts)
```

Comment:
I think these `self` references need to be `weak` to prevent a retain cycle.

I'm seeing `HomepageSettingsViewController` sticking around after leaving and it looks like a few copies are also made when visiting repeatedly:

<img width="891" alt="Screen Shot 2020-05-26 at 8 05 08 PM" src="https://user-images.githubusercontent.com/3250/82970815-711b2c00-9f8e-11ea-8cc5-1b8e4eb14311.png">